functions/composio_tools.py

```
@@ -262,6 +283,46 @@ def get_tools_for_toolkit(toolkit_slug: str) -> dict[str, dict[str, str]]:
+    from bots.models.convo_msg import Conversation
+    from routers.bots_api import api_hashids
+
+    manage_connections = {"enable": True, "callback_url": redirect_url}
+
+    if conversation_id:
+        try:
+            convo = Conversation.objects.get(
+                id=api_hashids.decode(conversation_id)[0]
+            )
+        except (IndexError, Conversation.DoesNotExist):
+            raise UserError(f"Conversation with id={conversation_id} not found")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)